src/main/java/org/kaazing/netx/ws/WebSocketFactory.java

```diff
@@ -222,6 +236,27 @@ public void setDefaultConnectTimeout(int connectTimeout) {
      }
 
     /**
+     * Sets the maximum payload length that is inherited by all the @{link WebSocket}s created using this factory.
+     * The maximum payload length can be Integer.MAX_VALUE - 14.
+     * <p>
+     * If this method is invoked after a connection has been successfully established, an IllegalStateException is thrown.
+     * If the maxPayloadLength <= 0 or maxPayloadLength > Integer.MAX_VALUE - 14, an IllegalArgumentException is thrown.
+     * <p>
+     * @param maxPayloadLength  maximum payload length for the connection
+     */
+    public void setMaxPayloadLength(int maxPayloadLength) {
+        if (maxPayloadLength > Integer.MAX_VALUE - 14) {
+            throw new IllegalArgumentException(format("Maximim payload length must not exceed %d", Integer.MAX_VALUE - 14));
+        }
```

Comment:
Hoist `Integer.MAX_VALUE - 14` as a constant, say `MAX_PAYLOAD_LENGTH_LIMIT` for improved readability.